pythran/pythonic/types/list.hpp

```diff
@@ -148,63 +148,52 @@ namespace pythonic {
                 static const size_t value = utils::nested_container_depth<list>::value;
                 static const bool is_vectorizable = true;
                 static const bool is_strided = false;
-                struct fake_shape {
-                  list const & the_list;
-                  fake_shape(list const & the_list) : the_list{the_list}{}
-                  template<class E> void init_shape(array<long, value>& res, E const& e, utils::int_<1>) const {
-                    res[value - 1] = e.size();
-                  }
-                  template<class E, size_t L> void init_shape(array<long, value>& res, E const& e, utils::int_<L>) const {
-                    res[value - L] = e.size();
-                    init_shape(res, e[0], utils::int_<L-1>{});
-                  }
-                  operator array<long, value>() const {
-                    array<long, value> res;
-                    init_shape(res, the_list, utils::int_<value>{});
-                    return res;
-                  }
-                } shape;
-
-
```

Comment:
The problem is that we can't add the list reference in this new struct as list is not fully declared.

The common interface for array like is still important IMO. I would prefer update this shape when we call append for example.